### ISSUE TITLE
Frozen Modal when starting payment for logged out user

### ISSUE THREAD

**Issue by pc-coholic** [iZettle/sdk-android#5]
# Expected Behavior

When starting a payment intent for a user that is not logged in, a `NotAuthorized` exception should be thrown - as explained in the `README`.

# Current Behavior

An empty modal is displayed with no way to close it. Only force-quitting the app will make it go away.

I would like to recommend to either actually raise the `NotAuthorized`-exception or - even better in my opinion - to just trigger the OAUTH login-process.

# Steps to Reproduce

````kotlin
        val paymentIntent = CardPaymentActivity.IntentBuilder(activity)
                .amount((receipt_total * BigDecimal("100.00")).toLong())  // TODO: Does not work for 0-digit currencies
                .reference(reference)
                .enableTipping(false)
                .build()

        activity.startActivityForResult(paymentIntent, REQUEST_CODE_EFT_PAYMENT)
````

Of course we can (and are) wrapping the `activity.startActivityForResult` in a check if the user is logged in - but this should not be necessary, imho.

## Context 

* Operating Version: Android 10
* SDK version: 1.4.0
* Originally reported as API-16526, reposted here to tracking purposes

**Comment by dmilovanovi** [comment]
Hey,

Thank you for reaching out to us.

Our team will have a look and we'll get back to you as soon as possible.


Kind regards.

**Comment by pvoid** [comment]
Fixed in version `1.8.3`
